src/Devices/Base/BaseDevice.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Transportation.Demo.Devices.Base
+{
+    public class BaseDevice
+    {
+        public List<SimulatedEvent> EventList = new List<SimulatedEvent>();
+
+        protected TransportationDeviceClient deviceClient;
+        private string connectionString;
+
+        public BaseDevice(string connectionString)
```

Comment:
For testability, make a ctor that takes DeviceClient as a parameter.

Reply:
@Sarah-Aly  not sure I understand how you want this changed. I understand what you're asking for, just nor sure about the implementation. Can you provide a link to a sample?

Reply:
@brentstineman https://github.com/brentstineman/Transit-IOTHub-Demo/blob/master/src/Functions/TransitFunctionApp/PurchaseTicketAction.cs 
        // Use the IInvokeDeviceMethod from the caller to allow for dependency injection.

        public PurchaseTicketAction(IInvokeDeviceMethod client, string payload, ILogger logger)

        {

            serviceClient = client;

            payloadMessage = payload;

            log = logger;

        }

Reply:
https://github.com/brentstineman/Transit-IOTHub-Demo/blob/master/src/Functions/TransitFunctionApp/PurchaseTicketAction.cs?fbclid=IwAR1k1rqlojXLOuQo-Jk4V5mUMsax96nt-ig3So98_xHhhRAhmhThkNkVODs#L17-L23

Reply:
I think I follow you. I'll need to create a mock for a DeviceToCloud interface then. I'll get this in today so you can review.